ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/messenger/attachmentpreview/AttachmentPreviewFragmentTest.java

```diff
@@ -0,0 +1,373 @@
+import android.content.Context;
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.v4.app.FragmentActivity;
+import android.support.v4.app.FragmentHostCallback;
+import android.support.v7.widget.PopupMenu;
+import android.text.TextUtils;
+import android.view.LayoutInflater;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import com.bumptech.glide.load.model.LazyHeaders;
+import com.kayako.sdk.android.k5.R;
+import com.kayako.sdk.android.k5.activities.KayakoAttachmentPreviewActivity;
+import com.kayako.sdk.android.k5.common.adapter.messengerlist.helper.AttachmentHelper;
+import com.kayako.sdk.android.k5.common.utils.ImageUtils;
+import com.kayako.sdk.android.k5.common.utils.file.FileAttachment;
+import com.kayako.sdk.android.k5.common.utils.file.FileAttachmentUtil;
+import com.kayako.sdk.android.k5.messenger.messagelistpage.helpers.DownloadAttachment;
+import com.kayako.sdk.android.k5.messenger.messagelistpage.helpers.FileAttachmentDownloadHelper;
+import com.kayako.sdk.auth.Auth;
+import com.kayako.sdk.auth.SessionAuth;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.powermock.reflect.Whitebox;
+
+import java.io.File;
+
+import static org.mockito.ArgumentMatchers.anyBoolean;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.reset;
+import static org.mockito.Mockito.verify;
+import static org.powermock.api.mockito.PowerMockito.doNothing;
+import static org.powermock.api.mockito.PowerMockito.doReturn;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.spy;
+import static org.powermock.api.mockito.PowerMockito.verifyStatic;
+import static org.powermock.api.mockito.PowerMockito.when;
+import static org.powermock.api.mockito.PowerMockito.whenNew;
+import static org.powermock.reflect.Whitebox.setInternalState;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({
+        ImageUtils.class,
+        LazyHeaders.class,
+        TextUtils.class,
+        LazyHeaders.Builder.class,
+        AttachmentPreviewFragment.class,
+        FileAttachmentUtil.class,
+        AttachmentHelper.class,
+        AttachmentHelper.AttachmentFileType.class
+
+})
+public class AttachmentPreviewFragmentTest {
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
```

Comment:
❗️ This snippet of code contains an instance variable assigned when declared, but not `final`. Consider marking this instance variable as `final` so, it should become a **final** instance variable. This way the code avoids intentionally/accidentally re-assignment of this instance variable and sustain the best OO practices.  

**[More details](https://confluence.devfactory.com/x/OKXNF)**
          
## The solution in this case
```java
@Rule 
public final ExpectedException expectedException = ExpectedException.none();
```
 

<a id="cb451650-4fa9-11e8-b7bf-a93e6485754a" href="none" codereview-uuid-reserved-tag />